pystac/collection.py

```diff
@@ -76,26 +90,34 @@ def clone(self) -> "SpatialExtent":
         Returns:
             SpatialExtent: The clone of this object.
         """
-        return SpatialExtent(deepcopy(self.bboxes))
+        return SpatialExtent(
+            bboxes=deepcopy(self.bboxes), extra_fields=deepcopy(self.extra_fields)
+        )
 
     @staticmethod
     def from_dict(d: Dict[str, Any]) -> "SpatialExtent":
-        """Constructs an SpatialExtent from a dict.
+        """Constructs a SpatialExtent from a dict.
 
         Returns:
             SpatialExtent: The SpatialExtent deserialized from the JSON dict.
         """
-        return SpatialExtent(bboxes=d["bbox"])
+        return SpatialExtent(
+            bboxes=d["bbox"], extra_fields={k: v for k, v in d.items() if k != "bbox"}
+        )
 
     @staticmethod
-    def from_coordinates(coordinates: List[Any]) -> "SpatialExtent":
+    def from_coordinates(
+        coordinates: List[Any], extra_fields: Optional[Dict[str, Any]] = None
+    ) -> "SpatialExtent":
         """Constructs a SpatialExtent from a set of coordinates.
 
         This method will only produce a single bbox that covers all points
         in the coordinate set.
 
         Args:
             coordinates : Coordinates to derive the bbox from.
+            extra_fields : Dictionary containing additional top-level fields defined on
+                the Spatial Extent object.
```

Comment:
```suggestion
                the SpatialExtent object.
```

Reply:
👍 Fixed in b0f9b496128f4ad50909812f8193401ccad9a924